gateway/gateway.go

```diff
@@ -310,16 +310,25 @@ func (gateway *HandleT) dbWriterWorkerProcess(process int) {
 		for _, userWorkerBatchRequest := range breq.batchUserWorkerBatchRequest {
 			jobList = append(jobList, userWorkerBatchRequest.jobList...)
 		}
-
-		if gwAllowPartialWriteWithErrors {
-			errorMessagesMap = gateway.jobsDB.StoreWithRetryEach(jobList)
-		} else {
-			err := gateway.jobsDB.Store(jobList)
-			if err != nil {
-				gateway.logger.Errorf("Store into gateway db failed with error: %v", err)
-				gateway.logger.Errorf("JobList: %+v", jobList)
-				panic(err)
+		err := gateway.jobsDB.WithStoreSafeTx(func(tx jobsdb.StoreSafeTx) error {
+			if gwAllowPartialWriteWithErrors {
+				errorMessagesMap = gateway.jobsDB.StoreWithRetryEachInTx(tx, jobList)
+			} else {
+				err := gateway.jobsDB.StoreInTx(tx, jobList)
+				if err != nil {
+					gateway.logger.Errorf("Store into gateway db failed with error: %v", err)
+					gateway.logger.Errorf("JobList: %+v", jobList)
+					return err
+				}
 			}
+
+			// rsources stats
+			rsourcesStats := rsources.NewStatsCollector(gateway.rsourcesService)
+			rsourcesStats.JobsStoredWithErrors(jobList, errorMessagesMap)
```

Comment:
if `gwAllowPartialWriteWithErrors` is `false`, we aren't reporting the numbers to sources?